packages/smooth_app/lib/data_models/product_list_supplier.dart

```diff
@@ -25,17 +25,22 @@ abstract class ProductListSupplier {
   /// Returns a helper supplier in order to refresh the data
   ProductListSupplier? getRefreshSupplier();
 
-  /// Clears the database and restarts from top page.
-  Future<void> clear() async {
+  /// Clears the database from page 2.
+  ///
+  /// Use-case: we have data on several pages, and the user wants to launch
+  /// the search again. That means
+  /// * all cached results are deprecated
+  /// * we start again a search from top page
+  /// * we should clear ALL pages
+  /// * but we clear only from page 2 and if the search didn't crash (server ko)
+  /// * in order to play it safe - when it crashes we still have data
+  Future<void> clearBeyondTopPage() async {
```

Comment:
```suggestion
  /// We should clear ALL pages, but we clear only 
  /// from page 2, so that if the request isn't 
  /// successful we still have data.
  Future<void> clearBeyondTopPage() async {
```

Reply:
@VaiTon Your rewriting means something slightly different, and slightly wrong.

In fact:
* if the request isn't successful, we still have ALL the "deprecated" data (all the pages)
* if the request is successful, as we reload from page 1, we remove the "deprecated" pages from page 2 (page 1 has just been refreshed and the other pages are now less meaningful as the product order probably changed)